### DISCUSSION TITLE
CXF jax-ws + jax-rs

### DISCUSSION THREAD

**Discussion by mjhaugsdal** [quarkiverse/quarkus-cxf#448] [Q&A] [answered]
Does quarkus-cxf support SOAP + REST endpoints together like CXF + Spring does?

In my spring context I can add both jaxrs and jaxws endpoints, giving me a site like:

Available SOAP services:
etc.
Endpoint address: ..
WSDL : ... 
Target namespace: .. 

Available RESTful services:
Endpoint address: http:// .. 

etc.

In my @Webservice annotated class I just annotate with both jaxrs and jaxws annotations.

**Comment by shumonsharif** [comment, marked as answer]
Hi @mjhaugsdal We do not support REST endpoints through this extension, because Quarkus already provides an extensive and reactive framework for JAX-RS.  It wasn't quite clear to me what your end goal is -- is it to leverage CXF JAX-RS, or to have a single page which lists the SOAP and REST endpoints?

I would also suggest you take a look at the links below, just in case you haven't already.

https://quarkus.io/guides/ [Search for JAX-RS or REST]
https://quarkus.io/guides/spring-web

**Reply by mjhaugsdal** [reply]
Thanks, was just curious

**Reply by mjhaugsdal** [reply]
@shumonsharif Just to clarify, we have an application today with spring that combines jax-rs and jax-ws like explained here: https://cxf.apache.org/docs/jax-rs-and-jax-ws.html
